exercises/02-publish-subscribe/after/Divergent.Customers/Program.cs

```diff
@@ -0,0 +1,37 @@
+using System;
+using System.Configuration;
+using System.Linq;
+using System.ServiceProcess;
+using System.Threading.Tasks;
+
+namespace Divergent.Customers
+{
+    static class Program
+    {
+        public async static Task Main(string[] args)
+        {
+            var host = new Host(ConfigurationManager.ConnectionStrings[Host.EndpointName].ToString());
+
+            // pass this command line option to run as a windows service
+            if (args.Contains("--run-as-service"))
+            {
+                using (var windowsService = new WindowsService(host))
+                {
+                    ServiceBase.Run(windowsService);
+                    return;
+                }
+            }
+
+            Console.Title = Host.EndpointName;
```

Comment:
afaik this crashes when you're running as a service, but I might be mistaken.

Reply:
When the endpoint is running as a service, the `--run-as-service` option should be passed, in which case this line will never be invoked.